cylc/flow/data_store_mgr.py

```diff
@@ -2291,10 +2291,11 @@ def delta_task_state(self, itask):
         update_time = time()
 
         # update task instance
-        tp_delta = self.updated[TASK_PROXIES].setdefault(
+        tp_delta: PbTaskProxy = self.updated[TASK_PROXIES].setdefault(
             tp_id, PbTaskProxy(id=tp_id))
         tp_delta.stamp = f'{tp_id}@{update_time}'
         tp_delta.state = itask.state.status
+        tp_delta.is_held = itask.state.is_held
```

Comment:
Currently:

| attr | updater |
| --- | --- |
| status | delta_task_state |
| is_held | delta_task_held |
| is_queued | delta_task_queued |
| is_runahead | delta_task_runahead |

I'm not sure why we would move `is_held` in with `delta_task_state`?

Reply:
Tome it looks like it makes sense to move all into `delta_task_state`? These are all handled by 1 method on `TaskProxy`, so it would reduce chances of forgetting to update them all in the data store

Reply:
We could do it all in one, but I think it would be *less* (not more) efficient for the data store to do this all at once as we would be pushing out all four attributes every time any one of them changes.

Reply:
If that's a concern, could do
```python
if tproxy.is_held != itask.state.is_held:
    tp_delta.is_held = itask.state.is_held
```
etc.

Or drop the commit as I've separated it out into its own commit now.

Reply:
Sounds good. I think, either we leave as is, or we combine all the related attrs and compare to previous value as you suggested.